msal_extensions/cache_lock.py

```diff
@@ -22,6 +22,7 @@ def __enter__(self):
         self._fh.write('{} {}'.format(pid, sys.argv[0]).encode('utf-8'))
 
     def __exit__(self, *args):
+        portalocker.unlock(self._fh)
```

Comment:
The change here is functionally correct. This can be our fall back option.

As we briefly mentioned in yesterday's talk, let's also try to see if we can build our `CrossPlatLock` on top of `portalocker.Lock`. Something like this. (Note that this change touches much more lines than our fall-back option, so we should test it even more carefully. But the benefit is we might be able to separate the behaviors in a clean way, once and for all. If it works, we would want to use it.)

```python
def __init__(self, lockfile_path):
    self._lockpath = lockfile_path
    self._lock = portalocker.Lock(lockfile_path, ...)

def __enter__(self):
    fh = self._lock.__enter__()
    fh.write('{} {}'.format(os.getpid(), sys.argv[0]).encode('utf-8'))
    return fh

def __exit__(self, *args):
    self._lock.__exit__(*args)
    try:
        # Attempt to delete the lockfile. In either of the failure cases enumerated below, it is
        # likely that another process has raced this one and ended up clearing or locking the
        # file for itself.
        os.remove(self._lockpath)
    except OSError as ex:
        if ex.errno != errno.ENOENT and ex.errno != errno.EACCES:
            raise
```